### ISSUE TITLE
Prevent overly steep beams

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#1677]
When beamed notes outside the system it happens that beams get far too steep. 
![steep_beams](https://user-images.githubusercontent.com/7693447/92927585-28ed2000-f43e-11ea-8d1a-a4c3a382b6a4.png)

According to Gould p. 20 end points of beams shouldn't be vertically further apart than 4 vu and and in shorter groups not more than 2 vu.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.1/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.1/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.1">
  <meiHead>
    <fileDesc>
      <titleStmt>
        <title>Prevent overly steep beam angles</title>
      </titleStmt>
      <pubStmt>
        <date isodate="2020">2020</date>
      </pubStmt>
      <seriesStmt>
        <title>Verovio test suite</title>
      </seriesStmt>
      <notesStmt>
        <annot>To avoid creating a steep beam angle the beam of shorter groups should not cross more than one staff-space.</annot>
      </notesStmt>
    </fileDesc>
    <encodingDesc>
      <appInfo>
        <application version="unknown">
          <name>Verovio</name>
        </application>
      </appInfo>
    </encodingDesc>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef>
            <staffGrp>
              <staffDef n="1" lines="5" clef.shape="G" clef.line="2"/>
            </staffGrp>
          </scoreDef>
          <section>
            <measure>
              <staff n="1">
                <layer>
                  <beam>
                    <note dur="8" oct="3" pname="d"/>
                    <note dur="8" oct="4" pname="f"/>
                  </beam>
                  <beam>
                    <note dur="8" oct="4" pname="d"/>
                    <note dur="8" oct="5" pname="f"/>
                  </beam>
                  <beam>
                    <note dur="8" oct="5" pname="d"/>
                    <note dur="8" oct="6" pname="f"/>
                  </beam>
                  <beam>
                    <note dur="8" oct="6" pname="d"/>
                    <note dur="8" oct="7" pname="f"/>
                  </beam>
                </layer>
              </staff>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

**Comment by craigsapp** [comment]
That is strange, since the current rule should be to attach stems to the middle staff line in such cases.  Having a beam/stem ending occurring within the ledger/leger line area should be strictly forbidden.

Here is what I am getting with the `develop-humdrum` branch, which was last merged with the `develop` branch a few weeks ago:

<img width="274" alt="Screen Shot 2020-09-11 at 6 11 46 AM" src="https://user-images.githubusercontent.com/3487289/92929757-e9024a80-f3f5-11ea-87e1-70571e65555e.png">

which is making me nervous about merging for issue https://github.com/rism-ch/verovio/issues/1653 😜 

Here is a test which is behaving correctly in VHV at the moment:

<img width="268" alt="Screen Shot 2020-09-11 at 6 17 16 AM" src="https://user-images.githubusercontent.com/3487289/92930138-85c4e800-f3f6-11ea-9531-3ce9cd70e442.png">


MEI data (the same as above, but changing one note down two octaves):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.1/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.1/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.1">
  <meiHead>
    <fileDesc>
      <titleStmt>
        <title>Prevent overly steep beam angles</title>
      </titleStmt>
      <pubStmt>
        <date isodate="2020">2020</date>
      </pubStmt>
      <seriesStmt>
        <title>Verovio test suite</title>
      </seriesStmt>
      <notesStmt>
        <annot>To avoid creating a steep beam angle the beam of shorter groups should not cross more than one staff-space.</annot>
      </notesStmt>
    </fileDesc>
    <encodingDesc>
      <appInfo>
        <application version="unknown">
          <name>Verovio</name>
        </application>
      </appInfo>
    </encodingDesc>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef>
            <staffGrp>
              <staffDef n="1" lines="5" clef.shape="G" clef.line="2"/>
            </staffGrp>
          </scoreDef>
          <section>
            <measure>
              <staff n="1">
                <layer>
                  <beam>
                    <note dur="8" oct="3" pname="d"/>
                    <note dur="8" oct="4" pname="f"/>
                  </beam>
                  <beam>
                    <note dur="8" oct="4" pname="d"/>
                    <note dur="8" oct="5" pname="f"/>
                  </beam>
                  <beam>
                    <note dur="8" oct="5" pname="d"/>
                    <note dur="8" oct="6" pname="f"/>
                  </beam>
                  <beam>
                    <note dur="8" oct="4" pname="d"/>
                    <note dur="8" oct="7" pname="f"/>
                  </beam>
                </layer>
              </staff>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

**Comment by lpugin** [comment]
I think a few things were broken in the beam refactoring merged a couple of days ago. I will post a few more issues.

**Comment by lpugin** [comment]
Rendering with 2.7.2
![image](https://user-images.githubusercontent.com/689412/92931565-ed555480-f443-11ea-891a-edf7da644e5c.png)

**Comment by craigsapp** [comment]
Is `regression` a new label?   I had not noticed it before...

**Comment by lpugin** [comment]
> Is `regression` a new label? I had not noticed it before...

Yes! Good to have when we can clearly identify one.

**Comment by craigsapp** [comment]
Verovio is reaching middle age 😉

**Comment by rettinghaus** [comment]
I can approve, that this came with #1666

**Comment by rettinghaus** [comment]
fixed with #1678
